packages/roosterjs-editor-plugins/lib/Paste/officeOnlineConverter/convertPastedContentFromWordOnline.ts

```diff
@@ -0,0 +1,233 @@
+    getFirstLeafNode,
+    getTagOfNode,
+    collapseNodes,
+    unwrap
+} from 'roosterjs-editor-dom';
+
+import ListItemBlock, { createListItemBlock } from './ListItemBlock';
+
+
+export function isWordOnlineWithList(node: HTMLElement): boolean {
+    return !!(node && node.querySelector(WORD_ONLINE_IDENTIFYING_SELECTOR));
+}
+
+// Word Online pasted content DOM structure as of July 12th 2019
+//<html>
+//  <body>
+//      <div class='OutlineGroup'>  ----------> this layer may exist depend on the content user paste
+//          <div class="OutlineElement">  ----------> text content
+//              <p></p>
+//          </div>
+//          <div class="ListItemWrapper">  ----------> list items: for unordered list, all the items on the same level is under the same wrapper
+//              <ul>                                       list items in the same list can be divided into different ListItemWrapper
+//                  <li></li>                              list items in the same list can also be divided into different Outline Group;
+//                  <li></li>
+//              </ul>
+//          </div>
+//      </div>
+//      <div class='OutlineGroup'>
+//          <div class="ListItemWrapper">  ----------> list items: for ordered list, each items has it's own wrapper
+//              <ol>
+//                  <li></li>
+//              </ol>
+//          </div>
+//          <div class="ListItemWrapper">
+//              <ol>
+//                  <li></li>
+//              </ol>
+//          </div>
+//      </div>
+//  </body>
+//</html>
+//
+
+/**
+ * Convert text copied from word online into text that's workable with rooster editor
+ * @param doc Document that is being pasted into editor.
+ */
+export default function convertPastedContentFromWordOnline(doc: HTMLDocument) {
+    const listItemBlocks: ListItemBlock[] = getListItemBlocks(doc);
+
+    listItemBlocks.forEach((itemBlock) => {
+
+        flattenListBlock(doc.body, itemBlock);
+
+        itemBlock.insertPositionNode = itemBlock.endElement.nextSibling;
+
+        let convertedListElement: Element;
+        itemBlock.listItemContainers.forEach((listItemContainer) => {
+            let listType: string = getContainerListType(listItemContainer); // list type that is contained by iterator.
+
+            // Initialize processed element with propery listType if this is the first element
+            if (!convertedListElement) {
+                convertedListElement = document.createElement(listType);
+            }
+
+            // Get all list items(<li>) in the current iterator element.
+            const currentListItems = listItemContainer.querySelectorAll('li');
+            currentListItems.forEach((item) => {
+                // If item is in root level and the type of list changes then
+                // insert the current list into body and then reinitialize the convertedListElement
+                const itemLevel = parseInt(item.getAttribute('data-aria-level'));
```

Comment:
who set the property "data-aria-level"? If it is Word Online, add comment to explain and make it a constant